oracle_cardano/processor/tx_processors/success/bridging_requested_processor_skyline.go

```diff
@@ -0,0 +1,355 @@
+	foundAnInvalidReceiverAddr := false
+
+	exchangeMargin := float64(0.000001)
+
+	exchangeRate, err := GetExchangeRate(tx.OriginChainID, metadata.DestinationChainID)
+	if err != nil {
+		return err
+	}
+
+	for _, receiver := range metadata.Transactions {
+		receiverAddr := strings.Join(receiver.Address, "")
+
+		if !cardanotx.IsValidOutputAddress(receiverAddr, cardanoDestConfig.NetworkID) {
+			foundAnInvalidReceiverAddr = true
+
+			break
+		}
+
+		var (
+			totalNativeCurrencyAmountToBridge uint64
+			totalWrappedTokenAmountToBridge   uint64
+		)
+
+		if receiverAddr == cardanoDestConfig.BridgingAddresses.FeeAddress {
+			if receiver.IsNativeTokenOnSrc || receiver.Additional != nil {
+				return fmt.Errorf("fee receiver metadata invalid: %v", metadata)
+			}
+
+			feeSum += receiver.Amount
+
+			continue
+		}
+
+		if receiver.Additional == nil {
+			receiver.Additional = &common.BridgingRequestMetadataCurrencyInfo{}
+		}
+
+		if receiver.IsNativeTokenOnSrc {
+			// amount_to_bridge must be >= minUtxoAmount on destination
+			if receiver.Amount < cardanoDestConfig.UtxoMinAmount {
+				foundAUtxoValueBelowMinimumValue = true
+
+				break
+			}
+
+			totalNativeCurrencyAmountToBridge = receiver.Additional.SrcAmount
+			totalWrappedTokenAmountToBridge = receiver.Amount
+		} else {
+			// amount_to_bridge must be >= minUtxoAmount on source
+			if receiver.Amount < cardanoSrcConfig.UtxoMinAmount {
+				foundAUtxoValueBelowMinimumValue = true
+
+				break
+			}
+
+			if receiver.Additional.DestAmount < cardanoDestConfig.UtxoMinAmount {
+				foundAUtxoValueBelowMinimumValue = true
+
+				break
+			}
+
+			totalNativeCurrencyAmountToBridge = receiver.Amount + receiver.Additional.SrcAmount
+		}
+
+		exchangeDiff := float64(receiver.Additional.DestAmount)*exchangeRate - float64(receiver.Additional.SrcAmount)
+		if (exchangeDiff > 0 && exchangeDiff > exchangeMargin) ||
+			(exchangeDiff < 0 && exchangeDiff < -1*exchangeMargin) {
+			return fmt.Errorf("found an exchange rate error in metadata: %v", metadata)
+		}
+
+		nativeCurrencyAmountSum.Add(nativeCurrencyAmountSum, new(big.Int).SetUint64(totalNativeCurrencyAmountToBridge))
+		wrappedTokenAmountSum.Add(wrappedTokenAmountSum, new(big.Int).SetUint64(totalWrappedTokenAmountToBridge))
+	}
+
+	if foundAUtxoValueBelowMinimumValue {
+		return fmt.Errorf("found a utxo value below minimum value in metadata receivers: %v", metadata)
+	}
+
+	if foundAnInvalidReceiverAddr {
+		return fmt.Errorf("found an invalid receiver addr in metadata: %v", metadata)
+	}
+
+	feeExchangeDiff := float64(metadata.FeeAmount.DestAmount)*exchangeRate - float64(metadata.FeeAmount.SrcAmount)
+	if (feeExchangeDiff > 0 && feeExchangeDiff > exchangeMargin) ||
+		(feeExchangeDiff < 0 && feeExchangeDiff < -exchangeMargin) {
+		return fmt.Errorf("found an exchage rate error if fee metadata: %v", metadata)
+	}
+
+	// update fee amount if needed with sum of fee address receivers
+	feeAmount := metadata.FeeAmount.SrcAmount + feeSum
```

Comment:
in reactor tx processor, it was:
metadata.FeeAmount.SrcAmount += feeSum
and thats how it should be here as well. we need to mutate the metadata, so that it can be used later when adding the claim.

The difference here, is that you will also need to update metadata.FeeAmount.DestAmount.
so do:
if feeSum > 0 {
metadata.FeeAmount.SrcAmount += feeSum
metadata.FeeAmount.DestAmount = metadata.FeeAmount.SrcAmount / exchangeRate
}

nativeCurrencyAmountSum.Add(nativeCurrencyAmountSum, new(big.Int).SetUint64(metadata.FeeAmount.SrcAmount))